docs/implementation.html

```diff
@@ -350,12 +350,28 @@
 /consumers/[group_id]/owners/[topic]/[partition_id] --> consumer_node_id (ephemeral node)
 </pre>
 
+<h4><a id="impl_clusterid" href="#impl_clusterid">Cluster Id</a></h4>
+
+<p>
+    The cluster id is an unique and immutable identifier assigned to a Kafka cluster. The cluster id is 22 characters long and the allowed characters is defined by the regular expression [a-zA-Z0-9_\-]+, which corresponds to the characters used by the URL-safe Base64 variant. It is auto-generated when the cluster is started for the first time.
+</p>
+
+<pre>
+/cluster/id  --> {"version":...,"id": ...} (persistent node)
+</pre>
+
 <h4><a id="impl_brokerregistration" href="#impl_brokerregistration">Broker node registration</a></h4>
 
 <p>
 The broker nodes are basically independent, so they only publish information about what they have. When a broker joins, it registers itself under the broker node registry directory and writes information about its host name and port. The broker also register the list of existing topics and their logical partitions in the broker topic registry. New topics are registered dynamically when they are created on the broker.
 </p>
 
+<h4><a id="impl_clusteridgeneration" href="#impl_clusteridgeneration">Cluster Id generation</a></h4>
```

Comment:
I think I'd just include this in the Cluster Id section, it doesn't seem worth it separating it.